### ISSUE TITLE
PROPOSAL - Add a REST endpoint for submissions to server

### ISSUE THREAD

**Issue by Fahad-pnw** [standup-raven/standup-raven#162]
**Is your feature request related to a problem? Please describe.**
Looking for a way to figure out how to submit daily stand ups to a server. Probably a POST request should do the trick and then i'd display a calendar like, nice preview to keep track on progress.
I'd like to work on this, if someone could guide a specific file / area where i'd write simple curl/js ajax calls to send to the server.

**Describe the solution you'd like**
add endpoint of your server API that accepts Post data...

**Describe alternatives you've considered**
none

**Additional context**
none

**Comment by harshilsharma63** [comment]
@Fahad-pnw sorry for the delay in responding.

The HTTP API endpoint for submitting a standup is located here - https://github.com/standup-raven/standup-raven/blob/593ce1b3f5d2895d36fe1c33557048cd0192f0ef/server/controller/standup.go#L32

I'm trying to understand your use case though. Are you trying to gather standup data externally and submit it in Standup Raven?

**Comment by Fahad-pnw** [comment]
If it is possible somehow, we'd like to create a calendar / post the data
to a database and then enlist each employee's daily stand up.

**Comment by harshilsharma63** [comment]
@Fahad-pnw are you trying to create a monthly report?

**Comment by Fahad-pnw** [comment]
Yes. A monthly, weekly, daily report sent off to a server script. This
script would ideally allow you to change date and view the reports for a
particular day in past.

**Comment by harshilsharma63** [comment]
@Fahad-pnw You can use the `executeGetStandup` API to fetch user's data https://github.com/standup-raven/standup-raven/blob/593ce1b3f5d2895d36fe1c33557048cd0192f0ef/server/controller/standup.go#L62

But there's no plan right now to push standup to external services. I do actually want to implement a report functionality as you've mentioned. Would you consider working on it?

**Comment by Fahad-pnw** [comment]
hi, yes, would be keen to work on it, probably will put on an intern to do
this, will update you if we do work on this any time soon :)

**Comment by harshilsharma63** [comment]
@Fahad-pnw that would be awesome. I'll post over here the ideas I had in mind.

**Comment by thatarchguy** [comment]
Any ideas on how to implement this?
Instead of a POST to a URL, I was thinking of being able to get the status report from the API to be scraped. 

Something like hitting this URL would display the report in json or whatever:
```
https://mattermost.example/plugins/standup-raven/report?channel_id=xx8490567xc&date=02-11-2021
```
I see this code, which could be used to generate it possibly:
https://github.com/standup-raven/standup-raven/blob/master/server/standup/notification/notification.go#L158

This issue is relevant as well
https://github.com/standup-raven/standup-raven/issues/198
